app/models/concerns/lockable.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module Lockable
+  def with_lock(lock_name, expire_after: 15.minutes, raise_on_failure: true)
+    with_redis do |redis|
```

Comment:
I suppose the main reason for this refactor is to use `with_redis` to enable reducing the number of connections in the pool down the line, but:
- this will check out a Redis client instance for the whole duration of the critical section, which might be unnecessarily long for this purpose
- with the code in this current PR, doesn't this mean checking out *more* Redis connections than the number of threads?
- eventually, if we change every Redis use site to check out from the pool and return into it, and we allow having less connections than threads in the pool, what happens when we run out of connections? The wait time becomes unpredictable, and/or we have to handle possible failures at every Redis use site

Reply:
The pool only allows checking out one connection per thread. Internally it just increments/decrements the reference counter on nested/repeated calls. So i think it’s a good start to rewriting the code to use redis connections only when needed.

Reply:
Oh ok, I missed that. In this case, my second point can be ignored. The two other points are still valid but that doesn't present a regression on the current code.